source/slang/lower-to-ir.cpp

```diff
@@ -1637,9 +1637,37 @@ struct StmtLoweringVisitor : StmtVisitor<StmtLoweringVisitor>
         SLANG_UNEXPECTED("`case` or `default` not under `switch`");
     }
 
-    void visitCompileTimeForStmt(CompileTimeForStmt*)
+    void visitCompileTimeForStmt(CompileTimeForStmt* stmt)
     {
-        SLANG_UNIMPLEMENTED_X("IR lowering of CompileTimeForStmt");
+        // The user is asking us to emit code for the loop
+        // body for each value in the given integer range.
+        // For now, we will handle this by repeatedly lowering
+        // the body statement, with the loop variable bound
+        // to a different integer literal value each time.
+        //
+        // TODO: eventually we might handle this as just an
+        // ordinary loop, with an `[unroll]` attribute on
+        // it that we would respect.
+
+        auto rangeBeginVal = GetIntVal(stmt->rangeBeginVal);
+        auto rangeEndVal = GetIntVal(stmt->rangeEndVal);
+
+        if (rangeBeginVal >= rangeEndVal)
+            return;
+
+        auto varDecl = stmt->varDecl;
+        auto varType = varDecl->type;
+
+        for (IntegerLiteralValue ii = rangeBeginVal; ii < rangeEndVal; ++ii)
+        {
+            auto constVal = getBuilder()->getIntValue(
+                varType,
+                ii);
```

Comment:
Yeah, why are these unrelated features showing up in the diff?!!

Reply:
I don't know what happened... I think at some point github is saying this PR is stale and I clicked "Update Branch" and then it automatically inserts a "merge" commit.